rest_framework/fields.py

```diff
@@ -493,6 +493,11 @@ def validate_empty_values(self, data):
         if data is None:
             if not self.allow_null:
                 self.fail('null')
+            # Nullable `source='*'` fields should not be skipped when its named
+            # field is given a null value. This is because `source='*'` means
+            # the field is passed the entire object, which is not null.
+            elif self.source == '*':
```

Comment:
the fix seems to be very simple on!! good job

Reply:
Wasn't the intention to return empty dict from validate_empty_values in this case? Otherwise I see failure in to_internal_value with error 'Invalid data. Expected a dictionary, but got NoneType.'?